pkg/operator/resource/resourceapply/core.go

```diff
@@ -328,24 +329,37 @@ func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, re
 }
 
 func SyncConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Recorder, sourceNamespace, sourceName, targetNamespace, targetName string, ownerRefs []metav1.OwnerReference) (*corev1.ConfigMap, bool, error) {
+	return SyncPartialConfigMap(client, recorder, sourceNamespace, sourceName, targetNamespace, targetName, nil, ownerRefs)
+}
+
+func SyncPartialConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Recorder, sourceNamespace, sourceName, targetNamespace, targetName string, syncedKeys sets.String, ownerRefs []metav1.OwnerReference) (*corev1.ConfigMap, bool, error) {
```

Comment:
could you rename `syncedKeys` to `sourceKeys` and move it right after `sourceName` param?

Reply:
I think the `syncedKeys` name is correct as those keys are going to appear in both the source and the target.

The fact that we only configure that in source of the destination->source map is an implementation detail that I would not like to surface to the function's interface. I can move the argument either to the front or right behind the `source, destination` argument pair, but I would not put in in the middle.